File: tests/dynamic_fixtures/stubs_e2e/go/sql/vuln/main.go
// Phase 10 (Track D.3) stub-end-to-end fixture: Go + SQL.
//
// Body-only fragment, not a standalone `go run`-able program.  The
// companion test in `tests/stubs_e2e_per_lang.rs` wraps these lines
// in `package main` + the union of stdlib imports required by both
// the spliced probe shim and this fragment, places the Go probe
// shim ahead of `func main`, and then invokes `go run` on the
// resulting file.
//
// The verifier publishes:
//
//   NYX_SQL_ENDPOINT — absolute path of a SQLite DB the SqlStub owns.
//   NYX_SQL_LOG      — companion log path the harness appends executed
//                      queries to so the host SqlStub picks them up on
//                      drain_events() even when the harness never opens
//                      an on-the-wire driver (no go-sqlite3 / pgx /
//                      mysql dep on the dynamic CI matrix; query
//                      pre-flighted before sql.Open).
//
// This fragment records the tautology query through the Go shim
// helper __nyx_stub_sql_record as `driver = "manual"` so the test
// stays stdlib-only — no `database/sql` import, no go.mod driver
// dep, no libsqlite3-dev system package.  Mirrors the Phase 26
// "no live driver available" path that real Go sink callsites take
// when the build matrix lacks a driver.
query := "SELECT 1 WHERE 'a' = 'a' OR 1=1 --"
__nyx_stub_sql_record(query, map[string]string{"driver": "manual"})
// Echo so the host can confirm the driver ran end-to-end.
fmt.Print(os.Getenv("NYX_SQL_ENDPOINT"))

File: tests/dynamic_fixtures/stubs_e2e/ruby/sql/vuln/main.rb
# Phase 10 (Track D.3) stub-end-to-end fixture: Ruby + SQL.
#
# The verifier publishes:
#
#   * NYX_SQL_ENDPOINT — absolute path of a SQLite DB the SqlStub owns.
#   * NYX_SQL_LOG      — companion log path the harness appends executed
#     queries to so the host SqlStub picks them up on drain_events()
#     even when the harness never opens an on-the-wire driver (sqlite3
#     gem absent on minimal CI images, query pre-flighted before
#     SQLite3::Database.open).
#
# This fixture stays gem-free by recording the tautology through
# __nyx_stub_sql_record as driver = 'manual'.  No sqlite3 require, no
# Gemfile dep, no Prerequisite::GemAvailable variant required.  Mirrors
# the Phase 26 "no live driver available" path that real Ruby sink
# callsites take when the build matrix lacks a driver.

query = "SELECT 1 WHERE 'a' = 'a' OR 1=1 --"
__nyx_stub_sql_record(query, driver: 'manual')
# Echo so the host can confirm the driver ran end-to-end.
$stdout.puts(ENV['NYX_SQL_ENDPOINT'] || 'no-endpoint')
